lib/shared/src/token/counter.ts

```diff
@@ -148,7 +154,7 @@ export class TokenCounter {
      * @returns The number of tokens in the message.
      */
     private static getTokenCountForMessage(message: Message): number {
-        if (!message) {
+        if (!message?.text) {
```

Comment:
Let's not do unexpected string to boolean coercion, this is biting us quite hard right now in https://github.com/sourcegraph/cody/pull/3734

```suggestion
        if (message?.text && message?.text.length > 0) {
```